article/article.md

```diff
@@ -0,0 +1,119 @@
+```code
+cy.visit(url);
+```
+After that, we fill all inputs with information.
+```code
+cy.get("input").eq(0).type("walter.bates");
+cy.get("input").eq(1).type("bpm");
+cy.get("input").eq(2).type("bpm");
+cy.get("input").eq(3).type("Walter");
+cy.get("input").eq(4).type("Bates");
+```
+Followed by modifying the password to not match the confirm password value.
+```code
+cy.get("input").eq(1).type("incorrect");
+```
+And the last action on the page being that of creating the user.
+```code
+cy.contains("button", "Create").click();
+```
+After we have setup this specific situation, we want to test that the "Passwords don't match." message is present on the page.
+```code
+cy.contains("p", message).should("be.visible");
+```
+After launching the test runner, we can see that the test is executed and is successful.
+![test passwords not matching](images/test_passwords_dont_match.jpg)
+
+## Testing the loader being displayed while the user is being created
+The scenario for this test is based on the display of a loader and checking that it disappears after five seconds.
+```code
+Given The server is started
+And API call response for "user created after delay" is mocked
+When I visit the index page
+And All inputs are filled
+And I try to create the user
+Then The loader is shown
+When I wait for 5000 delay
+Then I see the message about "User successfully created."
+And The loader is not shown
+```
+As you can see, some parts of this test are the same as the first one. The main things that are different are the definition of the API mock, checking that the loader is shown, waiting for 5 seconds and checking the end-state of the page.
+Mocking the API response after 5 seconds lets us have a moment during which the response has not arrived from the user API and thus the loading would be shown.
+The response is the same as the one that you would get from the Portal, which lets us be very close to reality.
+```code
+cy.fixture("json/userCreatedResponse.json").as("userCreated");
+cy.route({
+    method: "POST",
+    url: userAPI,
+    delay: 5000,
+    response: "@userCreated"
+}).as("userCreatedRoute");
+```
+While the user is created, we verify that the loader is shown.
+```code
+cy.get(".glyphicon.glyphicon-cog.gly-spin").should("be.visible");
+cy.contains("p", "Creating user.").should("be.visible");
+```
+After which we wait for 5 seconds.
+```code
+cy.wait(time);
+```
+And after 5 seconds, when the response arrives, we check that the message disappears.
+```code
+cy.get(".glyphicon.glyphicon-cog.gly-spin").should("not.be.visible");
+cy.contains("p", "Creating user.").should("not.be.visible");
+```
+
+As you can see, these tests don't take a long time to write and can be very powerful. 
+Awesome! Now you know how to test a form with Cypress! You can also check our [web-pages project](https://github.com/bonitasoft/bonita-web-pages) containing pages that we are currently developing.
```

Comment:
- how to test a "UI Designer" form ?
- Maybe the link should be https://github.com/bonitasoft/bonita-web-pages/tree/master/uid-pages as we don't want to advertise other parts of the repository